### DISCUSSION TITLE
Mouse cursor inconsistent in Firefox and Waybar

### DISCUSSION THREAD

**Discussion by ocharles** [YaLTeR/niri#360] [Q&A] [answered]
Hi all.

I have the following in my Niri config:

```kdl
cursor {
    xcursor-theme "breeze_cursors"
    xcursor-size 24
}
```

When my mouse cursor is over my desktop, or a `kitty` terminal my mouse cursor is Breeze, an expected. However, when I hover over a Firefox window, or my Waybar, I seem to have a default cursor that's smaller. Does anyone know what could be causing this?

**Comment by YaLTeR** [comment, marked as answer]
Firefox and Waybar use GTK 3, it probably has a different idea of where to load the cursors from than the XCURSOR env variable. You'll need to look into GTK 3 settings I guess

**Reply by ocharles** [reply]
Thanks, I didn't know that. Reading https://wiki.archlinux.org/title/Cursor_themes#GTK suggests I should look at `~/.config/gtk-3.0/settings.ini`, and indeed:

```
❯ cat ~/.config/gtk-3.0/settings.ini 
[Settings]
...
gtk-cursor-theme-name=Breeze_Snow
gtk-cursor-theme-size=24 
```

doesn't match Niri. I'll update that and make them consistent - thanks!

**Reply by ocharles** [reply]
Actually it seems that my GTK apps aren't loading the cursor at all - it's neither `breeze_cursors` _or_ `Breeze_Snow`. Nontheless I know this is a GTK think, so I'll accept this answer and do some digging.

**Reply by lpnh** [reply]
Maybe [nwg-look](https://github.com/nwg-piotr/nwg-look) would help you.

**Reply by ocharles** [reply]
Ah nice! I'm not sure what I changed but after tweaking that ini manually and using nwg-look, and probably restarting my session, I'm all sorted.

**Reply by kosayoda** [reply]
Hello from the future. To people in my future, `nwg-look` does work, but what `nwg-look` specifically does (that fixes @ocharles's issue) in addition to the `gtk-3.0` configuration is that [GNOME settings](https://wiki.archlinux.org/title/Cursor_themes#GNOME) are set:

Example:
```
$ gsettings set org.gnome.desktop.interface cursor-theme Bibata-Modern-Ice
$ gsettings set org.gnome.desktop.interface cursor-size 24
```

You shouldn't need to restart Firefox or your session, the changes should be instant.

**Reply by lpnh** [reply]
Yes! Hahaha, I learned that after a while. I've replaced `nwg-look` with [this script](https://github.com/lpnh/dotfiles/blob/main/home/.chezmoiscripts/run_once_after_10_setup-gtk-theme.sh) in my setup. By the way, thanks for sharing the information, it'll probably help someone else in another future.
